router/core/src/main/scala/io/buoyant/router/RoutingFactory.scala

```diff
@@ -10,7 +10,7 @@ object RoutingFactory {
   val description = "Performs per-request name binding"
 
   case class UnknownDst[Req](request: Req, cause: Throwable)
-    extends Exception(s"Unknown destination: ${cause.getMessage}", cause)
+    extends Exception(s"Unknown destination: $request\n${cause.getMessage}", cause)
```

Comment:
not a strong objection, but i'm slightly dubious about putting newlines in exception messages... i've worked on things in the past that try to parse exceptions from logs and this sort of things make it super painful.